python/lsst/ip/isr/calibType.py

```diff
@@ -0,0 +1,630 @@
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+    def toDict(self):
+        """Return a dictionary containing the calibration properties.
+
+        The dictionary should be able to be round-tripped through
+        `fromDict`.
+
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+
+        Raises
+        ------
+        NotImplementedError :
+            Raised if not implemented.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+    @classmethod
+    def fromTable(cls, tableList):
+        """Construct a calibration from a dictionary of properties.
+
+        Must be implemented by the specific calibration subclasses.
+
+        Parameters
+        ----------
+        tableList : `list` [`lsst.afw.table.Table`]
+            List of tables of properties.
+
+        Returns
+        ------
+        calib : `lsst.ip.isr.CalibType`
+            Constructed calibration.
+
+        Raises
+        ------
+        NotImplementedError :
+            Raised if not implemented.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+    def toTable(self):
+        """Return a list of tables containing the calibration properties.
+
+        The table list should be able to be round-tripped through
+        `fromDict`.
+
+        Returns
+        -------
+        tableList : `list` [`lsst.afw.table.Table`]
+            List of tables of properties.
+
+        Raises
+        ------
+        NotImplementedError :
+            Raised if not implemented.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+    def validate(self, other=None):
+        """Validate that this calibration is defined and can be used.
+
+        Parameters
+        ----------
+        other : object, optional
+            Thing to validate against.
+
+        Returns
+        -------
+        valid : `bool`
+            Returns true if the calibration is valid and appropriate.
+        """
+        return False
+
+    def apply(self, target):
+        """Method to apply the calibration to the target object.
+
+        Parameters
+        ----------
+        target : object
+            Thing to validate against.
+
+        Returns
+        -------
+        valid : `bool`
+            Returns true if the calibration was applied correctly.
+
+        Raises
+        ------
+        NotImplementedError :
+            Raised if not implemented.
+        """
+        raise NotImplementedError("Must be implemented by subclass.")
+
+
+class IsrProvenance(IsrCalib):
+    """Class for the provenance of data used to construct calibration.
+
+    Provenance is not really a calibration, but as we would like to
+    record this when constructing the calibration, and it provides an
+    example of the base calibration class.
+
+    Parameters
+    ----------
+    registryName : `str`, optional
```

Comment:
What is a registry name?

Reply:
I had intended this to be some way to identify the Butler repository that constructed the calibration, to ensure all the dataIds recorded could be transformed if needed.  This now seems unnecessary, as any Butler instance should get the same dataIds for the same data from astro_metadata_translator.
I'm changing this to instrument, as it still feels like we need some additional information in the provenance, and I think that's provided by that.